src/sentry/api/endpoints/release_thresholds/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Dict, List
+
+from snuba_sdk import Request as SnubaRequest
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import Condition, Op
+from snuba_sdk.entity import Entity
+from snuba_sdk.expressions import Granularity
+from snuba_sdk.function import Function
+from snuba_sdk.orderby import Direction, OrderBy
+from snuba_sdk.query import Query
+
+from sentry.snuba.dataset import Dataset
+from sentry.utils import snuba
+
+
+def get_errors_timeseries_counts_by_project_and_release(
+    start,
+    end,
+    project_id_list,
+    release_value_list,
+    organization_id,
+) -> List[Dict[str, Any]]:
+    query = Query(
+        match=Entity("events"),  # synonymous w/ discover dataset
+        select=[Function("count", [])],
+        groupby=[
+            Column("release"),
+            Column("project_id"),
+            Column("environment"),
+            Column("time"),  # groupby 'time' gives us a timeseries
+        ],
+        orderby=[
+            OrderBy(Column("time"), Direction.DESC),
+        ],
+        granularity=Granularity(60),
+        where=[
+            Condition(Column("type"), Op.EQ, "error"),
+            Condition(Column("timestamp"), Op.GTE, start),
+            Condition(Column("timestamp"), Op.LT, end),
+            Condition(Column("project_id"), Op.IN, project_id_list),
+            Condition(Column("release"), Op.IN, release_value_list),
+        ],
```

Comment:
Just realized that we aren't filtering on the environment here if it's filtered on we'll need to conditionally add it unless you can always ensure it's going to be defined

Reply:
ah nice good catch. will add env in here.